### ISSUE TITLE
reproducing

### ISSUE THREAD

**Issue by r-sayar** [Genentech/PerturbQA#1]
i tried reproducing the AUROC scores of the DE scores with the data provided, but I didn't get the same values. 

What I did: 
- download from zenodo
- ran load_de 
- ran the summer_parsing.load_outputs_summer
- ran auc_per_gene

the output hovered around .5 
k562 for example: 
Per-gene AUROC: 0.4978

What additional steps does one need to take?

**Comment by rmwu** [comment]
Can you confirm that you provided file paths to the files in `summer_outputs.zip`?

I would also recommend downloading `results.zip` which has per entry, pre-parsed dataframes available for inspection.

**Comment by qkrdmsghk** [comment]
> Can you confirm that you provided file paths to the files in `summer_outputs.zip`?
> 
> I would also recommend downloading `results.zip` which has per entry, pre-parsed dataframes available for inspection.

Hi, thank you for your great work!
I was wondering if the provided `summer_outputs.zip` corresponds directly to the test data of each dataset.

For example, I noticed that the size of `K562-de.json` in `summer_outputs` is different from that of the `K562-de test` dataset.

**Comment by r-sayar** [comment]
update: 
results.zip contents gets the correct results for the things I've tested

for each json in summer_outputs there are roughly as many lines as there are keys (

data = load_de(dataset)
X_test = data["test"]

keys = [(x["pert"], x["gene"]) for x in X_test] 
)
but there should be lines / 3 keys 


Also how did you do this step "To quantify uncertainty, we ran the retrieval step three times and report the average prediction over these three inference runs."  in your paper. E.g., did you take the one value which occurred the most often? If yes what did you do, if the most common was "-1" did you include it in the auroc?

**Comment by r-sayar** [comment]
i figured the issue out, in load_de it's loading from perturbqa/datasets/jurkat-de.csv but the json contains the train data of examples/sanity_check/data/20240712-nadig_jurkat.csv (this is just for the jurkat one but it should apply to all) 

would be interesting what the difference is between the those csvs 

this is the code I've used now, but I only get a .51/.52 for the jurkat dataset which in the paper gets .58
`#HAVE TO RUN FROM ROOT! 
from examples.summer import parsing as summer_parsing
from perturbqa.eval import auc_per_gene



# Configuration
DATASET = "jurkat"
TASK = 'de'
JSON_OUT = 'test_with_newly_downloaded_data/summer_outputs/jurkat-de.json'
RUN_SCRIPT = False  # set to True to run the long job from this notebook
SHOW_LAST_N = 20  # how many lines to print after run completes

import csv

#very bad naming, is because the other read_csv doesn't yield the correct path for whatever reason
def load_de_jurkat():
    """
    Load differential expression dataset
    """
    fp = f"examples/sanity_check/data/20240712-nadig_jurkat.csv"
    try:
        items = []
        with open(fp, 'r') as f:
            reader = csv.DictReader(f)
            data = []
            for item in reader:
                if "label" in item:
                    item["label"] = int(item["label"])
                data.append(item)
            items = data
    except:
        options = ["k562", "rpe1", "hepg2", "jurkat", "k562_set"]
        raise Exception(f"Invalid dataset: jurkat. Options: {options}")
    # organize by split
    data = {"train": [], "test": []}
    for item in items:
        if item["split"] not in data:
            continue
        data[item["split"]].append(item)
        del item["split"]
    return data

dataset = DATASET

data = load_de_jurkat()
X_test = data["test"]

keys = [(x["pert"], x["gene"]) for x in X_test]
true = [x["label"] for x in X_test]
keys_eval = keys

from collections import Counter

#NOT WORKING RN
key_val=summer_parsing.load_outputs_summer(JSON_OUT, keys_eval)

print(key_val[:10])

majority_votes = []
for sublist in key_val:
    if sublist:
        most_common = Counter(sublist).most_common(1)[0][0]
        majority_votes.append(most_common)
    else:
        majority_votes.append(None)  # or some default if empty

print(majority_votes[:4])

# Filter out entries where majority_vote is -1
filtered_keys_eval = []
filtered_majority_votes = []
filtered_true = []

for i in range(len(majority_votes)):
    if majority_votes[i] != -1:
        filtered_keys_eval.append(keys_eval[i])
        filtered_majority_votes.append(majority_votes[i])
        filtered_true.append(true[i])

# Now use the filtered lists
result = auc_per_gene(filtered_keys_eval, filtered_majority_votes, filtered_true)
print(result)




result = auc_per_gene(keys_eval, majority_votes, true)
print(result)`

**Comment by qkrdmsghk** [comment]
@r-sayar Thanks a lot! I finally figured out which file/split maps to the summer_outputs.zip and the data input:

As you mentioned, the author used `examples/sanity_check/data/20240712-replogle_K562_gw.csv` (same for others.)

The test split is `test`, which has `7,739` pert+gene samples.

In summer_outputs.zip there are `7,739 × 3 (may be times?) = 23,217` outputs for evaluation

But this still doesn’t exactly match the results.zip, which contains `23,212` pert+gene samples (may be from `perturbqa/datasets/K562-de.csv`).

**Comment by rmwu** [comment]
> Also how did you do this step "To quantify uncertainty, we ran the retrieval step three times and report the average prediction over these three inference runs." in your paper. E.g., did you take the one value which occurred the most often? If yes what did you do, if the most common was "-1" did you include it in the auroc?

Please see page 18: "A small fraction of inputs (p, g) resulted in no predictions after this filtering, or were unable to be parsed by our rule-based parsing. [...] Since these examples differed by model, we substituted the prediction with an uninformed baseline (the mean label of g over the training set) for evaluation."

> Re: parsing

1. Please use the `load_de(dataset)` labels for evaluation and benchmarking. These data statistics agree with those reported in the paper. The primary difference between this and the file you found is the label format.

2. I recommend reading from the `results.zip` via `results.ipynb` for the official results. The LLM generations seem to be only a subset (the first 7,739 x 3 lines), and it is a coincidence that the line numbers are similar. It is not possible to access the original files anymore since this work was completed during an internship.

**Comment by r-sayar** [comment]
1) why not filter them out?
2)1) i'll check this out, but the issue was today that there the Keys didnt match 2) okay this is good to know though, i think for some reason they are the exact same files as the ones which "temp.json....json" 

@qkrdmsghk it works precisely for the jurkat file, i think it should work for the others as well. but as @rmwu mentions, there is also a coincidence where the amount of entries/keys are extremely close (that goes for all datasets) but they domt match at all.

**Comment by rmwu** [comment]
> why not filter them out?

While the `SUMMER` method rarely abstains fully, the `LLM (No Retrieve)` baseline fully abstains in 2% of the DE and 36% (!) of Dir (Table 6). As a result, we couldn't filter them all out.
